integration-tests/src/test/java/com/linkedin/kafka/clients/largemessage/LargeMessageIntegrationTest.java

```diff
@@ -175,6 +176,87 @@ public void testLargeMessage() throws Exception {
+    }
+
+    long startTime = System.currentTimeMillis();
+    Properties props = new Properties();
+    props.setProperty("large.message.enabled", "true");
+    props.setProperty("max.message.segment.size", "200");
+    props.setProperty("client.id", "testProducer");
+    props.setProperty(LiKafkaProducerConfig.ENABLE_RECORD_HEADER_CONFIG, "true");
+
+    LiKafkaProducer<String, String> largeMessageProducer = createProducer(props);
+    Properties consumerProps = buildConsumerProps();
+    consumerProps.setProperty("auto.offset.reset", "earliest");
+    LiKafkaConsumer<String, String> consumer = createConsumer(consumerProps);
+
+        /* The test will send 100 different large messages to broker, consume from broker and verify the message contents.
+           Here for simplicity we use a large message segment as a large message, and chunk this */
+    Map<String, String> messages = new HashMap<>();
+    int numberOfLargeMessages = 100;
+    int largeMessageSize = 1000;
+    final Set<String> ackedMessages = new HashSet<>();
+    // Produce large messages.
+    for (int i = 0; i < numberOfLargeMessages; i++) {
+      final String messageId = LiKafkaClientsUtils.randomUUID().toString().replace("-", "");
+      String message = messageId + KafkaTestUtils.getRandomString(largeMessageSize);
+      messages.put(messageId, message);
+      largeMessageProducer.send(new ProducerRecord<>(TOPIC, message), (recordMetadata, e) -> {
+        // The callback should have been invoked only once.
+        assertFalse(ackedMessages.contains(messageId));
+        if (e == null) {
+          ackedMessages.add(messageId);
+        }
+      });
+    }
+    largeMessageProducer.close();
+    // All messages should have been sent.
+    assertEquals(ackedMessages.size(), messages.size());
+
+    // Consume and verify the large messages
+    List<TopicPartition> partitions = new ArrayList<>();
+    for (int i = 0; i < NUM_PARTITIONS; i++) {
+      partitions.add(new TopicPartition(TOPIC, i));
+    }
+    consumer.assign(partitions);
+
+    ConsumerRecords<String, String> records = null;
+    boolean consumptionStarted = false;
+    while (!consumptionStarted || records.count() > 0) {
+      records = consumer.poll(10);
+      if (records.count() > 0) {
+        consumptionStarted = true;
+      }
+      for (ConsumerRecord<String, String> consumerRecord : records) {
+        // Verify headers
+        Long eventTimestamp = LiKafkaClientsUtils.fetchTimestampHeader(consumerRecord.headers());
+        assertNotNull(eventTimestamp);
+        assertTrue(eventTimestamp >= startTime && eventTimestamp <= System.currentTimeMillis());
+        LargeMessageHeaderValue largeMessageHeaderValue = LiKafkaClientsUtils.fetchLargeMessageHeader(consumerRecord.headers());
+        assertEquals(largeMessageHeaderValue.getSegmentNumber(), -1);
+        assertEquals(largeMessageHeaderValue.getNumberOfSegments(), 6);
```

Comment:
nit:  the number of segments can be calculated based on the record size and max message size.